### ISSUE TITLE
Close JIRA tickets when linked github issue is 'closed' 

### ISSUE THREAD

**Issue by bpatelicent** [telicent-oss/jira-sync#13]
Noticed that https://github.com/telicent-oss/jira-sync/issues/3 is closed, but the JIRA ticket is still 'backlog'. 
I'm wondering if there is a way to mark a JIRA ticket as 'done/complete' if the associated github issue is 'closed'? 
Likewise, if an issue is reopened, the ticket should probably appear in the JIRA 'backlog' again.

**Comment by rvesse** [comment]
There probably is, there's a TODO for this lurking in the code already

Wasn't sure how far we wanted to go with automation because we might not always want someone closing something on the GitHub side to automatically close things on the JIRA side.  Especially since the person closing a ticket could be an outside contributor, they might be closing it because they've found a workaround, moved on, solved the problem another way etc but we might still want to address the original ticket.  Not sure if the GitHub API gives us enough context to tell who closed a ticket and trigger different actions for organisation members vs outside contributors

**Comment by bpatelicent** [comment]
Good point, can we trigger a status change to 'Ready for Review' in JIRA, serving as a reminder for one of us to go and check it and close/action as appropriate?
This may also need to work both ways, if I close a JIRA ticket - I'd probably want the linked issue to close too.

I've noticed a few things whilst setting up - I may turn this ticket into a 'snagging' list and we can see which of these we can action/won't bother with.

**Comment by rvesse** [comment]
> I've noticed a few things whilst setting up - I may turn this ticket into a 'snagging' list and we can see which of these we can action/won't bother with.

It would be preferable if we created each "snag" as a separate issue so that we can discuss them individually and decide which, if any, we want to address.

We can always have a central "snagging list" epic if that's helpful

**Comment by rvesse** [comment]
In the context of the new `--close-after-sync` option I added in PR #18 will also need to be careful that we don't prematurely close JIRA tickets in that case
